core-plugins/src/main/java/co/cask/hydrator/plugin/realtime/sink/RealtimeKinesisStreamSink.java

```
@@ -0,0 +1,199 @@
+    createAndWaitForStreamToBecomeAvailable(kinesisClient,
+                                            config.name,
+                                            config.shardCount);
+  }
+
+  @Override
+  public int write(Iterable<StructuredRecord> structuredRecords, DataWriter dataWriter) throws Exception {
+    int numRecordsWritten = 0;
+    for (StructuredRecord structuredRecord : structuredRecords) {
+      Schema schema = structuredRecord.getSchema();
+      Object data = structuredRecord.get(config.bodyField);
+      if (data == null) {
+        LOG.debug("Found null data. Skipping record.");
+        continue;
+      }
+      Schema.Field dataSchemaField = schema.getField(config.bodyField);
+      PutRecordRequest putRecordRequest = new PutRecordRequest();
+      putRecordRequest.setStreamName(config.name);
+      putRecordRequest.setPartitionKey(config.partition);
+      switch (dataSchemaField.getSchema().getType()) {
+        case BYTES:
+          numRecordsWritten += writeBytes(putRecordRequest, data);
+          break;
+        case STRING:
+          numRecordsWritten += writeString(putRecordRequest, data);
+          break;
+        default:
+          if (dataSchemaField.getSchema().getType().isSimpleType()) {
+            numRecordsWritten += writeBytes(putRecordRequest, String.valueOf(data));
+            break;
+          }
+          LOG.debug("Type {} is not supported for writing to stream", data.getClass().getName());
+          break;
+      }
+    }
+    return numRecordsWritten;
+  }
+
+  private int writeString(PutRecordRequest putRecordRequest, Object data) {
+    putRecordRequest.setData(ByteBuffer.wrap(Bytes.toBytes((String) data)));
+    try {
+      PutRecordResult result = kinesisClient.putRecord(putRecordRequest);
+      LOG.info("Data written at {} ", result.toString());
+      return 1;
+    } catch (Exception e) {
+      LOG.debug("could not write data to stream {}", putRecordRequest.getStreamName(), e);
```

I don't know if this is proper behavior.
Either we should propagate the exception, or log it at `WARN`.
@albertshau , suggestions?

---

due to the realtime nature of the sink, if we propagate error the subsequent data will also be not written to kinesis. changed to warning